sos/policies/auth/deviceauth.py

```diff
@@ -0,0 +1,313 @@
+        self._access_token = None
+        self._access_expires_at = None
+        self._refresh_token = None
+        self._refresh_expires_at = None
+        self._refresh_expires_in = None
+        self._user_verification_url = None
+        self.__device_code = None
+        self.AuthKey = Key()
+
+        # Lets check first if we have keyctl installed so we can
+        # store the token in the keyring
+        if not is_executable('keyctl'):
+            raise Exception("keyctl tool is not installed"
+                            " and is required to store auth tokens.")
+
+        self.client_identifier_url = client_identifier_url
+        self.token_endpoint = token_endpoint
+        self._use_device_code_grant()
+
+    def _use_device_code_grant(self):
+        """
+        Start the device auth flow. First check for the refresh token stored in
+        the session keyring. If they are not stored
+        or are invalid, request new device code. If the stored refresh token is
+        valid, use it to get new access_token
+
+        """
+        try:
+            stored_refresh_token = try_read_refresh_token()
+        except KeyRevokedError as e:
+            raise e
+        if not stored_refresh_token:
+            self._request_device_code()
+            print(
+                "Please visit the following URL to authenticate this"
+                f" device {self._verification_uri_complete}")
+            self.poll_for_auth_completion()
+        else:
+            self._use_refresh_token_grant(stored_refresh_token)
+
+    def _request_device_code(self):
+        """
+        Initialize new Device Authorization Grant attempt by
+        requesting a new device code.
+
+        """
+        data = "client_id={}".format(DEVICE_AUTH_CLIENT_ID)
+        headers = {'content-type': 'application/x-www-form-urlencoded'}
+        if not REQUESTS_LOADED:
+            raise Exception("python3-requests is not installed and is required"
+                            " for obtaining device auth token.")
+        try:
+            res = requests.post(
+                self.client_identifier_url,
+                data=data,
+                headers=headers)
+            res.raise_for_status()
+            response = res.json()
+            self._user_code = response.get("user_code")
+            self._verification_uri = response.get("verification_uri")
+            self._interval = response.get("interval")
+            self.__device_code = response.get("device_code")
+            self._verification_uri_complete = response.get(
+                "verification_uri_complete")
+        except requests.HTTPError as e:
+            raise requests.HTTPError("HTTP request failed "
+                                     "while attempting to acquire the tokens."
+                                     f"Error returned was {res.status_code} "
+                                     f"{e}")
+        except Exception as e:
+            raise e
```

Comment:
It was an attempt to capture exceptions that were not from requests, i.e. json ones. I can either make it more explicit or remove it completely if you think it's useless.

Reply:
Actually, lets remove it, because having these empty exceptions is not a good thing. If/When we get an uncaught exception in the future, we can add a explicit one.